### PR TITLE
[Deprecated] AzureML for Libero Fine-Tuning: Configs, MLflow Integration, and Documentation

### PR DESCRIPTION
**Pull request by noraabiakar**
This PR introduces a full AzureML pipeline for fine-tuning on the Libero dataset, including all necessary configuration files, documentation, and code changes to support MLflow tracking and improved reproducibility. 

## Technical Details
- AzureML YAMLs: Added `train.yaml`, `data-norm-stats.yaml`, and `dataset.yaml` for training and data normalization jobs.
- Documentation: Added `readme.md` with step-by-step instructions for setting up the environment, preparing data, and running jobs on AzureML.
- Dockerfile: Updated `serve_policy.Dockerfile` to install additional dependencies (ffmpeg, libsm6, libxext6) and MLflow/AzureML-MLflow Python packages.
- Training Script: Migrated from Weights & Biases (wandb) to MLflow for experiment tracking in `train.py`. All logging, hyperparameters, and metrics are now tracked with MLflow.
- Norm Stats Script: Enhanced `compute_norm_stats.py` to accept an explicit output path for norm stats, improving flexibility for AzureML jobs.
- Config: Changed the Libero data config in `config.py` to use local_files_only=True for local dataset handling.

### CODE REVIEW

File: aml/data-norm-stats.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    type: uri_folder
+    path: azureml://datastores/libero/paths/pi0_fast_libero/physical-intelligence/libero/
+  LEROBOT_HOME:
+    type: uri_folder
+    path: azureml://datastores/libero/paths/pi0_fast_libero/
+environment_variables:
+  LEROBOT_HOME: ${{outputs.LEROBOT_HOME}}
+display_name: pi0_compute_norm_stats
+experiment_name: compute_norm_stats
+compute: nabk-d48a-v4
```
Comment (jjottar) [inline aml/data-norm-stats.yaml]: Align with name of compute in README

File: aml/train.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandJob.schema.json
+command: uv run scripts/train.py pi0_fast_libero --exp-name=${{inputs.experiment_name}} --overwrite --assets-base-dir ${{outputs.ASSETS_BASE}} --checkpoint-base-dir ${{outputs.MODEL_CHECKPOINTS}} --num-workers 8 --fsdp-devices 1 --batch-size 32
```
Comment (jjottar) [inline aml/train.yaml]:
```suggestion
command: uv run scripts/train.py ${{inputs.config_name}} --exp-name=${{inputs.experiment_name}} --overwrite --assets-base-dir ${{outputs.ASSETS_BASE}} --checkpoint-base-dir ${{outputs.MODEL_CHECKPOINTS}} --num-workers 8 --fsdp-devices 1 --batch-size 32
```

File: aml/readme.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    ```
+    curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | sudo bash
+    sudo apt-get install git-lfs
+    ```
+* Use git-lfs to download the dataset (preferrably in the /mnt directory of your VM, to avoid space issues):
+    
+    ```
+    (optional) sudo mkdir -p /mnt/data && cd /mnt/data
+    git lfs install
+    git clone https://huggingface.co/datasets/physical-intelligence/libero
```
Comment (jjottar) [inline aml/readme.md]:
```suggestion
    sudo git clone https://huggingface.co/datasets/physical-intelligence/libero
```

**Review by jjottar — COMMENTED** [review]
minor

File: aml/readme.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    ```
+
+* Copy the libero dataset to the "libero" container in the "libero" storage account under the path "pi0_fast_libero/physical-intelligence/libero".
+   Login with azcopy first. If you are using a VM with managed identity you can use:
+   ```
+   azcopy login --identity
+   ```
+   Copy the files
+   ```
+   azcopy "/path/to/downloaded/libero/dataset/*" "https://libero.blob.core.windows.net/libero/pi0_fast_libero/physical-intelligence/libero/" --recursive=true
```
Comment (jjottar) [inline aml/readme.md]:
```suggestion
   azcopy copy "/path/to/downloaded/libero/dataset/*" "https://libero.dfs.core.windows.net/libero/pi0_fast_libero/physical-intelligence/libero/" --recursive=true
```

**Review by jjottar — COMMENTED** [review]
minor

File: examples/fine-tune-with-azureml/readme.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    - Users and VM identities need "Storage blob data contributor" role
+    - The "libero" container in the "libero" storage account needs to be added as an Azure ML datastore (make sure to mount it as ADLS Gen2 with HNS otherwise you'll face `Os.Errors` during checkpoint writing!)
+
+## Create the AzureML environment
+
+First, create the AzureML environment. Here and in the following steps it's called it `libero-env`.
+```
+ az ml environment create --name libero-env --build-context . --dockerfile-path scripts/docker/serve_policy.Dockerfile --resource-group  robotics-ch-north --workspace-name robotics-ch-north
+```
+
```
Comment (jjottar) [inline examples/fine-tune-with-azureml/readme.md:22]:
```suggestion
Note: if you trigger this command from an AzureVM, make sure to first run `sudo az extension add --upgrade -n ml`, as it will require the up-to-date ml extension. Likewise, make sure you have write access (e.g. the Storage Blob Data Contributor role) to the storage account associated to the AzureML workspace, as it will be required by this command.
```

**Review by jjottar — COMMENTED** [review]
Suggesting to add clarifying comments for creating the AML environment from Azure VM

File: aml/readme.md (hunk 3 above)
Comment (noraabiakar) [reply aml/readme.md]: I think this depends on the environment. Usually git clone doesn't require sudo.

File: examples/fine-tune-with-azureml/readme.md (hunk 5 above)
Comment (noraabiakar) [reply examples/fine-tune-with-azureml/readme.md:22]: The storage blob data contributor is listed in the requred resources. Is it not clear enough? 
For the upgrade of the extension, I think we would actually need to say to install az cli, login and install the ml extension, but this might be too much info. I think we either need to assume some basic azure/azureml knowledge or we need more detailed docs altogether. What do you think?

Comment (jjottar) [reply examples/fine-tune-with-azureml/readme.md:22]: But all that comes installed by default on the AzureML VMs no? (note I started the paragraph saying "if you trigger from an AzureVM")

 I didn't have to install the az or any of that. I created a new VM and then just did
`sudo az extension add --upgrade -n ml`
